src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -166,32 +175,179 @@ public void CompilerDataLogger_Dispose_ShouldReadSarifV1()
+            BinaryAnalyzerContext context = CreateTestContext();
+            ExecutionException exception = new ExecutionException(type: "test", message: "testExecutionException", stackTrace: "testStackTrace", innerException: new Exception());
+            AnalysisSummary summary = new AnalysisSummary();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.WriteException(exception, summary);
+            sendItems.Count.Should().Be(1);
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_CreateScvOutputFile_ShouldSkipIfPathIsNull()
+        {
+            BinaryAnalyzerContext context = CreateTestContext();
+            var fileSystem = new Mock<IFileSystem>();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.CreateCsvOutputFile(null, false);
+            sendItems.Count.Should().Be(0);
+            context.Dispose();
+        }
+
+        private List<ITelemetry> TestSetup(string sarifLogFilePath, BinaryAnalyzerContext context, Sarif.SarifVersion sarifVersion, out CompilerDataLogger logger, IFileSystem fileSystem = null, bool isDisabledLogger = false)
+        {
+            List<ITelemetry> sendItems = null;
+            sendItems = new List<ITelemetry>();
+
+            if (isDisabledLogger)
+            {
+                logger = new CompilerDataLogger(sarifLogFilePath, sarifVersion, context ?? new BinaryAnalyzerContext(), fileSystem);
+            }
+            else
+            {
+                TelemetryClient telemetryClient;
+                TelemetryConfiguration telemetryConfiguration;
+                telemetryConfiguration = new TelemetryConfiguration();
+                telemetryConfiguration.InstrumentationKey = Guid.NewGuid().ToString();
+                telemetryConfiguration.TelemetryChannel = new StubTelemetryChannel { OnSend = item => sendItems.Add(item) };
+                telemetryConfiguration.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+
+                telemetryClient = new TelemetryClient(telemetryConfiguration);
+
+                CompilerDataLogger.s_injectedTelemetryClient = telemetryClient;
+                CompilerDataLogger.s_injectedTelemetryConfiguration = telemetryConfiguration;
+
+                context.Policy = new Sarif.PropertiesDictionary();
+                logger = new CompilerDataLogger(sarifLogFilePath, sarifVersion, context ?? new BinaryAnalyzerContext(), fileSystem);
+            }
 
             return sendItems;
         }
 
+        private BinaryAnalyzerContext CreateTestContext(string outputPath = null)
+        {
+            string csvOutputPath = outputPath ?? @$"C:\temp\{Guid.NewGuid()}.sarif";
```

Comment:
>@$"C:\temp\{Guid.NewGuid()}.sarif" [](http://example.com/codeflow?start=49&length=34)

just a note if we want to change it to OS unrelated path btw

Reply:
Path.Combine